docs/user-guide.md

```diff
@@ -341,6 +343,7 @@ won't need to know.
 
```

Comment:
I think `Type annotation` hasn't been introduced before, but maybe that's intuitive enough.
And maybe "Type use annotations` to be more precise?

Reply:
Sadly we can never get away with saying "type annotation" since ElementType uses `TYPE` (completely improperly!) just to mean `CLASS`.

I would prefer "type-usage annotation" even though it doesn't roll of the tongue... because "type use" is just wrong. (An example "use" of `boolean` is "to indicate whether a mutating method actually mutated", and an example "usage" of `boolean` is "as the return type of `Collection.add`".)

Reply:
The annotations are called `TYPE_USE` in the `java.lang.annotation` API so I think we can reasonably call them "type-use annotations" and be understood. I actually think "type-usage annotations" would be slightly harder to understand.

Reply:
I can live with that.